source/ceylon/net/uri/Query.ceylon

```diff
@@ -1,26 +1,13 @@
-import ceylon.collection {
-    LinkedList
-}
-
 "Represents a URI Query part"
 by("Stéphane Épardaud")
-shared class Query(Parameter* initialParameters) {
-    
-    "The list of query parameters"
-    shared LinkedList<Parameter> parameters 
-            = LinkedList<Parameter>();
-    
-    for(p in initialParameters) {
-        parameters.add(p);
-    }
+shared class Query {
 
-    "Adds a query parameter"
-    shared void add(Parameter param)
-            => parameters.add(param);
+    "The list of query parameters"
+    shared Parameter[] parameters;
 
-    "Adds a single raw (percent-encoded) query parameter, where name and value have to be parsed"
-    shared void addRaw(String part)
-            => add(parseParameter(part));
+    shared new Query(Parameter* parameters) {
+        this.parameters = parameters.sequence();
```

Comment:
Whoa, did I really write that :smile:? Actually, I remember thinking the arg was an `Iterable`, which would be mutable..